coordinator/core/marbleapi.go

```diff
@@ -315,33 +315,68 @@ func (c *Core) generateMarbleAuthSecrets(req *rpc.ActivationReq, marbleUUID uuid
 	return authSecrets, nil
 }
 
-func (c *Core) setTTLSConfig(marble manifest.Marble, secrets reservedSecrets) error {
+func (c *Core) setTTLSConfig(marble manifest.Marble, specialSecrets reservedSecrets, userSecrets map[string]manifest.Secret) error {
 	if len(marble.TLS) == 0 {
 		return nil
 	}
 
-	ttlsConf := make(map[string]map[string]map[string]string)
-	ttlsConf["tls"] = make(map[string]map[string]string)
+	// Wrap the secrets to have the ".Marblerun" / ".Secrets" prefix in template notation
+	secretsWrapped := secretsWrapper{
+		Marblerun: specialSecrets,
+		Secrets:   userSecrets,
+	}
+
+	ttlsConf := make(map[string]map[string]map[string]map[string]string)
+	ttlsConf["tls"] = make(map[string]map[string]map[string]string)
+	ttlsConf["tls"]["Incoming"] = make(map[string]map[string]string)
+	ttlsConf["tls"]["Outgoing"] = make(map[string]map[string]string)
 
 	pemCaCert := pem.Block{Type: "CERTIFICATE", Bytes: c.intermediateCert.Raw}
 	stringCaCert := string(pem.EncodeToMemory(&pemCaCert))
 
-	pemClientCert := pem.Block{Type: "CERTIFICATE", Bytes: secrets.MarbleCert.Cert.Raw}
+	pemClientCert := pem.Block{Type: "CERTIFICATE", Bytes: specialSecrets.MarbleCert.Cert.Raw}
 	stringClientCert := string(pem.EncodeToMemory(&pemClientCert))
 
-	pemClientKey := pem.Block{Type: "PRIVATE KEY", Bytes: secrets.MarbleCert.Private}
+	pemClientKey := pem.Block{Type: "PRIVATE KEY", Bytes: specialSecrets.MarbleCert.Private}
 	stringClientKey := string(pem.EncodeToMemory(&pemClientKey))
 
 	for _, tag := range marble.TLS {
+		connConf := make(map[string]string)
```

Comment:
I'd move this back in the inner loop(s) because otherwise you have to be careful that one config doesn't contain a value from another config. E.g., incoming user-defined may have a cacrt.